.github/workflows/unit-tests.yml

```diff
@@ -305,6 +305,48 @@ jobs:
       group: ${{ github.event_name == 'push' && github.run_number || github.ref }}-musl
       cancel-in-progress: true
 
+  ASAN_UBSAN:
+    runs-on: ubuntu-22.04
+    timeout-minutes: 90
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
+
+    - name: Set up Conda
+      uses: ./.github/actions/setup-conda
+      with:
+        environment-file: ci/deps/actions-311-numpydev.yaml
```

Comment:
Did you chose this file since you wanted a minimal environment? If so might as well create a new env file with a stable version of numpy